service/cluster/kubernetes_cluster_test.go

```diff
@@ -0,0 +1,64 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"context"
+	"testing"
+
+	controllerv1beta1 "github.com/percona-platform/dbaas-api/gen/controller"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+func TestKubernetesClusterServiceCheckConnection(t *testing.T) {
+	t.Run("Wrong kube config", func(t *testing.T) {
+		logrus.SetLevel(logrus.DebugLevel)
+		i18nPrinter := message.NewPrinter(language.English)
+		k := NewKubernetesClusterService(i18nPrinter)
+		kubeConfig := `{
+			"kind": "Config",
+			"apiVersion": "v1",
+			"preferences": {},
+			"clusters": [
+				{
+					"name": "minikube",
+					"cluster": {
+						"server": "https://1.2.3.4:8443",
+					}
+				}
+			],
+			"contexts": [
+				{
+					"name": "minikube",
+					"context": {
+						"cluster": "minikube",
+						"user": "minikube"
+					}
+				}
+			],
+			"current-context": "minikube"
+		}`
```

Comment:
Why we use JSON instead of YAML?

Reply:
Why not? We were using `-o json`, that's why I used the same format.

Reply:
We use JSON for K8s API requests and responses, but pretty much everyone uses YAML for kubeconfig file. We can reasonably expect people to provide kubeconfig in YAML format. _If_ both JSON and YAML work now, then we can keep, but otherwise we should not force users to convert YAML to JSON (and using YAML will cover JSON too since YAML 1.2 is JSON superset).

Reply:
I'll check if YAML works and if it is, will create separate test for it as well

Reply:
Just a note.
YAML should work as well. The issue could be to pass it from frontend to pmm-managed; YAML inside JSON, escaping not enough. Maybe a form- or base64- encoding/deconning is required

Reply:
Checked it and it works :)

Reply:
So `kubectl` always uses YAML parser for `kubeconfig` file, and since JSON is a subset of YAML 1.2, it just works

Reply:
But I think we should use YAML both internally (in all tests, etc) and externally (in documentation, etc) simply because it is more common and tests more